appimage/build_appimage.sh

```diff
@@ -179,38 +247,108 @@ generate_ssl_cert() {
 }
 
 build_frontend() {
-    pushd "$ISLAND_PATH/cc/ui" || handle_error
-    npm install sass-loader node-sass webpack --save-dev
-    npm update
+  pushd "$ISLAND_PATH/cc/ui" || handle_error
+
+  log_message "Generating front end"
+  npm ci
+  npm run dist
+
+  popd || handle_error
 
-    log_message "Generating front end"
-    npm run dist
-    popd || handle_error
+  remove_node_modules
+}
+
+remove_node_modules() {
+  # Node has served its purpose. We don't need to deliver the node modules with
+  # the AppImage.
+  rm -rf "$ISLAND_PATH"/cc/ui/node_modules
 }
 
 add_monkey_icon() {
-	unlink "$APPDIR"/python.png
-	mkdir -p "$APPDIR"/usr/share/icons
-	cp "$REPO_MONKEY_SRC"/monkey_island/cc/ui/src/images/monkey-icon.svg "$APPDIR"/usr/share/icons/infection-monkey.svg
-	ln -s "$APPDIR"/usr/share/icons/infection-monkey.svg "$APPDIR"/infection-monkey.svg
+  unlink "$APPDIR"/python.png
+  mkdir -p "$APPDIR"/usr/share/icons
+  cp "$1"/monkey_island/cc/ui/src/images/monkey-icon.svg "$APPDIR"/usr/share/icons/infection-monkey.svg
+  ln -s "$APPDIR"/usr/share/icons/infection-monkey.svg "$APPDIR"/infection-monkey.svg
 }
 
 add_desktop_file() {
-	unlink "$APPDIR"/python3.7.9.desktop
-	cp ./infection-monkey.desktop "$APPDIR"/usr/share/applications
-	ln -s "$APPDIR"/usr/share/applications/infection-monkey.desktop "$APPDIR"/infection-monkey.desktop
+  unlink "$APPDIR/python${PYTHON_VERSION}.desktop"
+  cp ./infection-monkey.desktop "$APPDIR"/usr/share/applications
+  ln -s "$APPDIR"/usr/share/applications/infection-monkey.desktop "$APPDIR"/infection-monkey.desktop
 }
 
 add_apprun() {
-	cp ./AppRun "$APPDIR"
+  cp ./AppRun "$APPDIR"
 }
 
 build_appimage() {
-    log_message "Building AppImage"
-    ARCH="x86_64" appimagetool "$APPDIR"
+  log_message "Building AppImage"
+  ARCH="x86_64" appimagetool "$APPDIR"
+  apply_version_to_appimage "$1"
 }
 
-if is_root; then
+apply_version_to_appimage() {
+  log_message "Renaming Infection_Monkey-x86_64.AppImage -> Infection_Monkey-$1-x86_64.AppImage"
+  mv "Infection_Monkey-x86_64.AppImage" "Infection_Monkey-$1-x86_64.AppImage"
+}
+
+as_root=false
+monkey_repo="$DEFAULT_REPO_MONKEY_HOME"
+monkey_version="dev"
+agent_binary_dir=""
+branch="develop"
+
+while (( "$#" )); do
+case "$1" in
+  --agent-binary-dir)
+    if [ -n "$2" ] && [ "${2:0:1}" != "-" ]; then
+      agent_binary_dir=$2
+      shift 2
+    else
+      missing_argument "$1"
+    fi
+    ;;
+  --as-root)
+    as_root=true
+    shift
+    ;;
+  --branch)
+    if [ -n "$2" ] && [ "${2:0:1}" != "-" ]; then
+      branch=$2
+      shift 2
+    else
+      missing_argument "$1"
+    fi
+    ;;
+  -h|--help)
+    echo_help
+    ;;
+  --monkey-repo)
+    if [ -n "$2" ] && [ "${2:0:1}" != "-" ]; then
+      monkey_repo=$2
+      shift 2
+    else
+      missing_argument "$1"
+    fi
+    ;;
+  --version)
+    if [ -n "$2" ] && [ "${2:0:1}" != "-" ]; then
+      monkey_version=$2
+      shift 2
+    else
+      missing_argument "$1"
+    fi
+    ;;
+  -*)
```

Comment:
why do we need the dash?

Reply:
Since we don't have any positional arguments, we don't. Good catch.